File: mols2grid_widget/js/lib/widget.js
var widgets = require('@jupyter-widgets/base');
var _ = require('lodash');

var CommWidgetModel = widgets.DOMWidgetModel.extend({
    defaults: _.extend(widgets.DOMWidgetModel.prototype.defaults(), {
        _model_name : 'CommWidgetModel',
        _view_name : 'CommWidgetView',
        _model_module : 'mols2grid_widget',
        _view_module : 'mols2grid_widget',
        _model_module_version : '0.1.0',
        _view_module_version : '0.1.0',
        grid_id : "default",
        selection : "{}",
        callback_args : "{}"
    })
});


var CommWidgetView = widgets.DOMWidgetView.extend({
    // Defines how the widget gets rendered into the DOM
    render: function() {
        let grid_id = this.model.get('grid_id');
        let name = "_MOLS2GRID_" + grid_id;
        window[name] = this.model;
        // Python -> JS
        this.model.on('change:grid_id', this.value_changed, this);
    },

    value_changed: function() {
        
    }
});


module.exports = {
    CommWidgetModel: CommWidgetModel,
    CommWidgetView: CommWidgetView
};

File: mols2grid_widget/mols2grid_widget/widget.py
from ipywidgets import register, DOMWidget
from traitlets import Unicode

@register
class CommWidget(DOMWidget):
    """Custom widget to handle comm between the grid and the Python backend"""
    _view_name = Unicode('CommWidgetView').tag(sync=True)
    _model_name = Unicode('CommWidgetModel').tag(sync=True)
    _view_module = Unicode('mols2grid_widget').tag(sync=True)
    _model_module = Unicode('mols2grid_widget').tag(sync=True)
    _view_module_version = Unicode('^0.1.0').tag(sync=True)
    _model_module_version = Unicode('^0.1.0').tag(sync=True)

    grid_id = Unicode("default").tag(sync=True)
    selection = Unicode("{}").tag(sync=True)
    callback_args = Unicode("{}").tag(sync=True)
